flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -638,6 +638,14 @@ public ProgramTargetDescriptor executeUpdate(String sessionId, String statement)
 			result.close();
 			// catch everything such that the query does not crash the executor
 			throw new SqlExecutionException("Invalid SQL query.", t);
+		} finally {
+			// Remove the temporal table object.
+			context.wrapClassLoader(() -> {
+				// Quotes the table name because as a SqlIdentifier, it is invalid.
+				context.getTableEnvironment()
+						.dropTemporaryTable(String.format("`%s`", jobName));
```

Comment:
How about just use a `UUID` for table name? The quoting logic seems a bit hack to me.

Reply:
This table name is temporary, so we just need to give a name that is not conflict with what user defines. Change to '_tmp_table_hashCode(query)'